src/network/netdev/tunnel.c

```diff
@@ -126,6 +127,16 @@ static int netdev_sit_fill_message_create(NetDev *netdev, Link *link, sd_netlink
                         return log_netdev_error_errno(netdev, r, "Could not append IFLA_IPTUN_6RD_PREFIXLEN attribute: %m");
         }
 
+        if (t->isatap >= 0) {
+                uint16_t flags = 0;
+
+                SET_FLAG(flags, SIT_ISATAP, t->isatap);
+
+                r = sd_netlink_message_append_u16(m, IFLA_IPTUN_FLAGS, flags);
```

Comment:
What happens `t->isatap == 0`? If I understand correctly, then `t->isatap == 0` and `t->isatap == -1` have no difference. If so, treating `t->isatap` as tristate is meaningless. How about the following?
```
typedef struct Tunnel {
        ...
        bool isatap;
        ...
} Tunnel;
```
```
if (t->isatap) {
        r = sd_netlink_message_append_u16(m, IFLA_IPTUN_FLAGS, SIT_ISATAP);
        if (r < 0)
                return log_netdev_error_errno(...);
}
```

Reply:
> What happens `t->isatap == 0`? If I understand correctly, then `t->isatap == 0` and `t->isatap == -1` have no difference. If so, treating `t->isatap` as tristate is meaningless. How about the following?

 when isatap is ```0```  sit tunnel does not turn on the flag while ```1``` does.
Basically we are reusing the code of sit tunnel. Hence  

```
sudo ip tu ad isatap mode isatap local 10.7.7.7 remote 10.8.8.8 

ip -d link show sittun 
11: sittun@NONE: <POINTOPOINT,NOARP> mtu 1480 qdisc noop state DOWN mode DEFAULT group default qlen 1000
    link/sit 10.7.7.7 peer 10.8.8.8 promiscuity 0 
    sit ip6ip remote 10.8.8.8 local 10.7.7.7 ttl inherit pmtudisc 6rd-prefix 2002::/16 addrgenmode eui64 numtxqueues 1 numrxqueues 1 gso_max_size 65536 gso_max_segs 65535 

```
now with let's create a isatap
```
$ sudo ip tunnel add is0 mode isatap local 192.0.2.1 ttl 64
$ ip -d link show is0
12: is0@NONE: <NOARP> mtu 1480 qdisc noop state DOWN mode DEFAULT group default qlen 1000
    link/sit 192.0.2.1 brd 0.0.0.0 promiscuity 0 
    sit ip6ip remote any local 192.0.2.1 ttl 64 pmtudisc isatap 6rd-prefix 2002::/16 addrgenmode eui64 numtxqueues 1 numrxqueues 1 gso_max_size 65536 gso_max_segs 65535 
```

Reply:
Ohh I forgot we create sit and turn on the flag kernel does accepts it and ISATAP flag gets turned on.

Reply:
Sorry, I cannot understand your explanation. I've tested `ISATAP=false` and unsetting `ISATAP=`. Then, I find no difference between them. So, my question is, is it necessary to store `ISATAP=` setting as a tristate? I guess it is enough to store as a boolean.